### ISSUE TITLE
Ruby 3 and backwards compatibility

### ISSUE THREAD

**Issue by cgoettel** [apalmblad/ruby-shadow#28]
@apalmblad I'm starting a new issue so we can track everything here.

In the meantime, I think it's important if we decide if we even need to maintain backwards compatibility. Ruby 1.9 has been obsolete since February 23, 2015. Ruby 2.2 has been obsolete since April 1, 2018. I think we should only worry about being compatible with Ruby 2.3 and on. Anyone who needs to use a previous version of Ruby can just use ruby-shadow 2.5.0. Yes?

That said, Ruby 3 is backwards compatible with Ruby 2, but Ruby 2 is not backwards compatible with Ruby 1.9. I'm looking through the codebase and updating it for Ruby 3 compatibility. You can assign this issue to me, if you'd like.

**Comment by cgoettel** [comment]
`ruby extconf.rb` works on Ruby 3.0.0. Not a full test, but a good first step.

**Comment by apalmblad** [comment]
Yes, I think it's reasonable to drop support for obsolete rubies.  Per https://www.ruby-lang.org/en/downloads/, seems like even the 2.5 series is deprecated.

Thank for taking on some work to improve ruby 3 compatibility - even dropping the def of RUBY19 and the initial breaking extconf.rb checks would be nice.

I was able to build in ruby 3 on OS X.
